tool/tsh/db.go

```diff
@@ -726,12 +734,16 @@ func onDatabaseConnect(cf *CLIConf) error {
 	return nil
 }
 
-// getDatabaseInfo fetches information about the database from tsh profile is DB is active in profile. Otherwise,
+// getDatabaseInfo fetches information about the database from tsh profile if DB is active in profile. Otherwise,
 // the ListDatabases endpoint is called.
 func getDatabaseInfo(cf *CLIConf, tc *client.TeleportClient, dbName string) (*tlsca.RouteToDatabase, types.Database, error) {
-	database, err := pickActiveDatabase(cf)
+	dbRoute, err := pickActiveDatabase(cf)
 	if err == nil {
-		return database, nil, nil
+		db, err := getDatabase(cf, tc, dbRoute.ServiceName)
```

Comment:
The `db, err := getDatabase(cf, tc, dbRoute.ServiceName) method connects to the teleport proxy and  calls the  API so it will add additionally delay to the tsh db connect execution mentioned in  one of the bellow comments.

Reply:
types.Database is required for RoleSet.EnumerateDatabase* functions. Is there a way to get that type without api call?

Reply:
Unfortunately no. We are storing some of database information  in  tsh database user profile but it is not complete list. For `getDatabase ` we should avoid calling ConnectToProxy inside `getDatabase`